Content.Shared/Explosion/Components/OnIgniteTimerTriggerComponent.cs

```diff
@@ -0,0 +1,54 @@
+using Robust.Shared.Audio;
+using Robust.Shared.GameStates;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Content.Shared.Explosion.Components
+{
+    [RegisterComponent, NetworkedComponent]
+    public sealed partial class OnIgniteTimerTriggerComponent : Component
+    {
+        [DataField] public float Delay = 1f;
+
+        /// <summary>
+        ///     If not null, a user can use verbs to configure the delay to one of these options.
+        /// </summary>
+        [DataField] public List<float>? DelayOptions = null;
+
+        /// <summary>
+        ///     If not null, this timer will periodically play this sound while active.
+        /// </summary>
+        [DataField] public SoundSpecifier? BeepSound;
+
+        /// <summary>
+        ///     Time before beeping starts. Defaults to a single beep interval. If set to zero, will emit a beep immediately after use.
+        /// </summary>
+        [DataField] public float? InitialBeepDelay;
+
+        [DataField] public float BeepInterval = 1;
+
+        /// <summary>
+        ///     Whether the timer should instead be activated through a verb in the right-click menu
+        /// </summary>
+        [DataField] public bool UseVerbInstead = false;
+
+        /// <summary>
+        ///     Should timer be started when it was stuck to another entity.
+        ///     Used for C4 charges and similar behaviour.
+        /// </summary>
+        [DataField] public bool StartOnStick;
+
+        /// <summary>
+        ///     Allows changing the start-on-stick quality.
+        /// </summary>
+        [DataField("canToggleStartOnStick")] public bool AllowToggleStartOnStick;
+
+        /// <summary>
+        ///     Whether you can examine the item to see its timer or not.
+        /// </summary>
+        [DataField] public bool Examinable = true;
+    }
+}
```

Comment:
This looks copy-pasted from OnuseTimerTrigger. Components should be made reuseable.

This should just be a generic "TriggerOnIgniteComponent" instead with all these copied options removed.

Reply:
I could probably remove some of these options, but why not leave some of them like beepinterval and beepsound? Wouldn't this allow for more items to use this component in the future? Same for below, IO could probably remove a bit, but why not leave some things the same if they work?

Reply:
@metalgearsloth Can I get your thoughts on this?

Reply:
They should be on 1 component not spread across multiple, it just increases technical debt and makes it more of a pain to maintain the code.

Reply:
just have AutomaticTriggerComponent or whatever i called it then call TriggerSystem's public api to start the timer once its igniter